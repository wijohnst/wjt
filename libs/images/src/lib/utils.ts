import { readFileSync } from 'fs';

/**
 * Accepts a path string and returns the contents of the file as a buffer
 * @param {string} path
 * @returns {Promise<Buffer>}
 */
export const getBufferFromPath = async (path: string): Promise<Buffer> => {
  return readFileSync(path);
};

/**
 * Checks if the substring is a CDN image using the provided regular expression
 * @param {string} path
 * @param {RegExp} matcher
 * @returns {boolean}
 */
export const isCdnImage = (path: string, matcher: RegExp): boolean => {
  return matcher.test(path);
};
