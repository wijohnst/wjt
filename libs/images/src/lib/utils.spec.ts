import mock from 'mock-fs';
import { getBufferFromPath, isCdnImage } from './utils';

describe('utils', () => {
  describe('getBufferFromPath', () => {
    beforeEach(() => {
      mock({
        mock_images: {
          '200_200_mock.jpg': 'content',
        },
      });
    });

    afterEach(() => {
      mock.restore();
    });

    afterAll(() => {
      mock.restore();
    });

    test('should be defined', () => {
      expect(getBufferFromPath).toBeDefined();
    });

    test('should return a buffer from a file path', async () => {
      const result = await getBufferFromPath('mock_images/200_200_mock.jpg');

      expect(result).toBeDefined();
      expect(result).toMatchSnapshot();
    });
  });
});

describe('images', () => {
  describe('isCdnImage', () => {
    test('should be defined', () => {
      expect(isCdnImage).toBeDefined();
    });

    test('should return true for a CDN image', () => {
      const path =
        'https://wjt.sfo2.cdn.digitaloceanspaces.com/sample_image.webp';
      const matcher = /https:\/\/wjt\.sfo2\.cdn\.digitaloceanspaces\.com\/.*/;

      expect(isCdnImage(path, matcher)).toBe(true);
    });
  });
});
