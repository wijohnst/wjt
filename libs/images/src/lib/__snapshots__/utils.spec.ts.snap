// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getBufferFromPath should return a buffer from a file path 1`] = `
{
  "data": [
    99,
    111,
    110,
    116,
    101,
    110,
    116,
  ],
  "type": "Buffer",
}
`;
